Date: 2016-06-07 12:42:40 +0200

From: 31f05de9c4cd04b3ff9bb6b3030d24be3be837d4 2016-06-17 18:27:31 +0200

Subject: [PATCH] RDK-17052: set library installation directory from libdir env variable

	As we are working in Yocto environment, it is better to set install dir from Yocto
	env variables. The same goes for CFLAGS.

Source: GITHUB


Upstream-Status: Submitted [ http://git.yoctoproject.org/cgit/cgit.cgi/meta-ivi/commit/meta-ivi/recipes-support-ivi/keyutils/keyutils/0001-keyutils-Use-libdir-from-the-Yocto-environment.patch?id=31f05de9c4cd04b3ff9bb6b3030d24be3be837d4 ]

Signed-off-by: Frederico Cadete <frederico.cadete@awtce.be>
Signed-off-by: Clement Dransart <clement.dransart@awtce.be>
---

 Makefile | 30 ++++--------------------------
 1 file changed, 4 insertions(+), 26 deletions(-)

diff --git a/Makefile b/Makefile
index c904eaf..22b74ba 100644
--- a/Makefile
+++ b/Makefile
@@ -51,32 +51,11 @@ LIBNAME		:= libkeyutils.so.$(APIVERSION)
 
 ###############################################################################
 #
-# Guess at the appropriate lib directory and word size
+# Use libdir from the Yocto environment
 #
 ###############################################################################
-ifeq ($(origin LIBDIR),undefined)
-LIBDIR		:= $(shell ldd /usr/bin/make | grep '\(/libc\)' | sed -e 's!.*\(/.*\)/libc[.].*!\1!')
-endif
-ifeq ($(origin USRLIBDIR),undefined)
-USRLIBDIR	:= $(patsubst /lib/%,/usr/lib/%,$(LIBDIR))
-endif
-BUILDFOR	:= $(shell file /usr/bin/make | sed -e 's!.*ELF \(32\|64\)-bit.*!\1!')-bit
-
-LNS		:= ln -sf
-
-ifeq ($(origin CFLAGS),undefined)
-ifeq ($(BUILDFOR),32-bit)
-CFLAGS		+= -m32
-LIBDIR		:= /lib
-USRLIBDIR	:= /usr/lib
-else
-ifeq ($(BUILDFOR),64-bit)
-CFLAGS		+= -m64
-LIBDIR		:= /lib64
-USRLIBDIR	:= /usr/lib64
-endif
-endif
-endif
+LIBDIR    := $(libdir)
+USRLIBDIR := $(libdir)
 
 ###############################################################################
 #
@@ -161,7 +140,7 @@ endif
 	$(INSTALL) -D $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
 	$(LNS) $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SONAME)
 	mkdir -p $(DESTDIR)$(USRLIBDIR)
-	$(LNS) $(LIBDIR)/$(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
+	$(LNS) $(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
 	$(INSTALL) -D keyctl $(DESTDIR)$(BINDIR)/keyctl
 	$(INSTALL) -D request-key $(DESTDIR)$(SBINDIR)/request-key
 	$(INSTALL) -D request-key-debug.sh $(DESTDIR)$(SHAREDIR)/request-key-debug.sh
@@ -271,6 +250,5 @@ show_vars:
 	@echo APIVERSION=$(APIVERSION)
 	@echo LIBDIR=$(LIBDIR)
 	@echo USRLIBDIR=$(USRLIBDIR)
-	@echo BUILDFOR=$(BUILDFOR)
 	@echo SONAME=$(SONAME)
 	@echo LIBNAME=$(LIBNAME)
-- 
1.9.1

